File: 2.1-4 binary-adding/code.js
function binaryAdd(A, B){
    let n = A.length;
    let C = [];
    let carriedBit = 0;
    let i = 0;

    // Assume the binary bits are stored in an array starting with the least important bit
    // ending with the most important bit.
    for(; i < n; i++){
        let bitA = A[i];
        let bitB = B[i];
        let sum = bitA + bitB + carriedBit;
        let bitC = sum % 2;
        carriedBit = Math.floor(sum / 2);
        C[i] = bitC;
    }
    
    C[i] = carriedBit;
    return C;
}

File: 2.1-4 binary-adding/code.py
def binaryAdd(A, B):
    n = len(A)
    C = []
    carriedBit = 0
    i = 0
    
    for i in range(n):
        bitA = A[i]
        bitB = B[i]
        sum = bitA + bitB + carriedBit
        bitC = sum % 2
        carriedBit = sum // 2
        C.append(bitC)
    
    C.append(carriedBit)
    return C
